### ISSUE TITLE
`unique` will sort the input when `sort=False` on cuda

### ISSUE THREAD

**Issue by fuzzyswan** [pytorch/pytorch#88447]
### 🐛 Describe the bug

`unique` will sort the input when `sort=False` on cuda. On contrast, it will not sort on cpu

```py
import torch

input = torch.FloatTensor([1, 3])
torch.unique(input, sorted=False)
# tensor([3., 1.])
torch.unique(input.cuda(), sorted=False)
# tensor([1., 3.], device='cuda:0')
```

### Versions

torch: 1.13.0
cuda: 11.6

**Comment by ngimel** [comment]
That's expected behavior, and docs mention that.

**Comment by fuzzyswan** [comment]
Hi @ngimel, the doc of `unique` mention that "when dim is specified, torch.unique always sort the tensor at the beginning". In this case, the `dim` is not specified, so it should not sort the tensor

**Comment by fuzzyswan** [comment]
Hi @ngimel, actually in this case, the order of input tensor is ascending. 

The CPU will reverse the tensor, which is not expected since it should remain the order of the input tensor with `sorted=False` (even the `sort=True`, it should return `[1, 3]` instead of `[3, 1]`

**Comment by jgong5** [comment]
cc @min-jean-cho

**Comment by ngimel** [comment]
I would still say it's unclear documentation. When `sort=False` it means that output can be in any order, not that the input order will be preserved. When `sorted=True` output is guaranteed to be in sorted order. We'll accept a doc PR.

**Comment by jordane95** [comment]
The current implementation of returning random order is quite misleading... Is there any plan on implementing original order perserving features when `sorted=False`?

**Comment by gkneighb** [comment]
No longer reproduces on torch 2.11.0 (CPU on Apple Silicon arm64). `torch.unique(torch.FloatTensor([1, 3]), sorted=False)` returns `[1., 3.]` as expected; also tested `[1, 2, 3, 4, 5]` and inputs with duplicates — all preserve order or correctly deduplicate. Looks safe to close.
